app/docs/md/patterns/event-listeners.md

```diff
@@ -0,0 +1,191 @@
+
+For example:
+
+- A user submits a form
+- A user clicks on a button
+- A web page finishes loading
+- etc.
+
+**[Learn more about events](https://developer.mozilla.org/en-US/docs/Web/Events)**
+
+## Adding a click event listener
+
+Suppose we have a simple web component called `custom-button` that we want to add a click event listener to.
+
+<doc-code filename="app/components/custom-button.mjs">
+
+```javascript
+import CustomElement from '@enhance/custom-element'
+
+export default class CustomButton extends CustomElement {
+  render ({ html, state }) {
+    return html`
+        <style>
+            button {
+                border: 1px solid white;
+                padding: 2px;
+            }
+        </style>
+        <button>Click me!</button>
+    `
+  }
+}
+
+customElements.define('custom-button', CustomButton)
+```
+
+</doc-code>
+
+As a best practice, event listeners for web components should be added inside the `connectedCallback` function, since this method runs when the element is added to the DOM.
+
+Inside our `connectedCallback` method, we'll use the `querySelector` method on the custom element, to get the button inside our component. Then we will add a `click` listener to the button.
+
+<doc-code filename="app/components/custom-button.mjs">
+
+```javascript
+import CustomElement from '@enhance/custom-element'
+
+export default class CustomButton extends CustomElement {
+  connectedCallback () {
+    this.button = this.querySelector('button')
+    this.button.addEventListener('click', function (event) {
+      alert('I was clicked')
+    })
+  }
+
+  render ({ html, state }) {
+    return html`
+        <style>
+            button {
+                border: 1px solid white;
+                padding: 2px;
+            }
+        </style>
+        <button>Click me!</button>
+    `
+  }
+}
+
+customElements.define('custom-button', CustomButton)
+```
+
+</doc-code>
+
+That works great but if we truly do want to follow best practices we should clean up our event listeners when our component is removed from the DOM. We'll remove that event listener in the `disconnectedCallback` method. In order to do that we'll need to refactor our component slightly.
+
+1. Extract our click handler to a method named `clickHander`.
+1. Save a reference to the `button` element in the `constructor`.
+1. Bind `this` to the callback function.
+1. Add the event listener in `connectedCallback`.
+1. Remove the event listener in `disconnectedCallback`.
+
+<doc-code filename="app/components/custom-button.mjs">
+
+```javascript
+import CustomElement from '@enhance/custom-element'
+
+export default class CustomButton extends CustomElement {
+  handleClick (event) {
+    alert('I was clicked')
+  }
```

Comment:
This is SUPER personal preference I reckon, but I'd move `handleClick` to follow after the constructor. I feel like if I don't see the constructor first in a WC it's a bit odd. Feel free to ignore!